payment-method-messaging/src/main/java/com/stripe/android/paymentmethodmessaging/view/PaymentMethodMessagingViewModel.kt

```diff
@@ -34,36 +33,33 @@ internal class PaymentMethodMessagingViewModel @Inject constructor(
     fun loadMessage() {
         viewModelScope.launch {
             _messageFlow.update {
-                try {
-                    val message = stripeApiRepository.retrievePaymentMethodMessage(
-                        paymentMethods = config.paymentMethods.map { it.value },
-                        amount = config.amount,
-                        currency = config.currency,
-                        country = config.countryCode,
-                        locale = config.locale.toLanguageTag(),
-                        logoColor = config.imageColor?.value ?: if (isSystemDarkThemeProvider()) {
-                            PaymentMethodMessagingView.Configuration.ImageColor.Light.value
-                        } else {
-                            PaymentMethodMessagingView.Configuration.ImageColor.Dark.value
-                        },
-                        requestOptions = ApiRequest.Options(config.publishableKey),
-                    )
-                    if (
-                        message == null ||
-                        message.displayHtml.isBlank() ||
-                        message.learnMoreUrl.isBlank()
-                    ) {
-                        Result.failure(Exception("Could not retrieve message"))
+                retrievePaymentMethodMessage().mapCatching { message ->
+                    if (message.displayHtml.isBlank() || message.learnMoreUrl.isBlank()) {
+                        throw Exception("Could not retrieve message")
```

Comment:
Keeping existing behavior here. Should we just throw an `APIConnection` instead?

Reply:
🤷 Either way is fine by me